### DISCUSSION TITLE
Similar crate like this for python type hint

### DISCUSSION THREAD

**Discussion by jhoobergs** [Aleph-Alpha/ts-rs#150] [Ideas]
I would like to create something similar to this but instead of typescript I would like to generate python type hint. For the structs I would use [TypedDict](https://peps.python.org/pep-0589/) and unions for the enums. This seems doable.

The code of this crate seems like a perfect starting point to me, but I am wondering what would be the best way to do this. I see the following options:
1. I create a fork of this repo and change everything ts related to python type hints.
2. I create a fork of this repo and make the current code more general as to support both TS and Python and create a PR to this repo.

The first option is somewhat easier, but makes it almost impossible to update to changes in this repo.
The second option is a bit more work, but might be interesting to do, as this generalization will open the door the easily add bindings for other languages.

Would you be open to the second option?  If you see other options, please let me know.

**Comment by escritorio-gustavo** [comment]
I think option 1 is a more sensible solution. This crate's goal is to help integration with TypeScript, so maintaining Python type hints is not a thing we'd be able to keep doing.

**Reply by escritorio-gustavo** [reply]
I mean... I don't even know Python so I'd be useless in maintenance lol
